go-controller/pkg/node/gateway_udn.go

```diff
@@ -146,18 +158,56 @@ func NewUserDefinedNetworkGateway(netInfo util.NetInfo, networkID int, node *v1.
 		kubeInterface: kubeInterface,
 		// Generate a per network conntrack mark to be used for egress traffic.
 		masqCTMark: ctMarkUDNBase + uint(networkID),
-	}
+		gateway:    gw,
+	}, nil
 }
 
 // AddNetwork will be responsible to create all plumbings
 // required by this UDN on the gateway side
 func (udng *UserDefinedNetworkGateway) AddNetwork() error {
-	return udng.addUDNManagementPort()
+	err := udng.addUDNManagementPort()
+	if err != nil {
+		return fmt.Errorf("failed to add management port %s; error: %v", udng.GetNetworkName(), err)
+	}
+	if udng.openflowManager != nil {
+		udng.openflowManager.addNetwork(udng.NetInfo, udng.masqCTMark)
+
+		waiter := newStartupWaiter()
+		readyFunc := func() (bool, error) {
+			if err := setBridgeNetworkOfPorts(udng.openflowManager.defaultBridge, udng.GetNetworkName()); err != nil {
+				return false, fmt.Errorf("failed to set network %s's openflow ports for default bridge; error: %v", udng.GetNetworkName(), err)
+			}
+			if udng.openflowManager.externalGatewayBridge != nil {
+				if err := setBridgeNetworkOfPorts(udng.openflowManager.externalGatewayBridge, udng.GetNetworkName()); err != nil {
+					return false, fmt.Errorf("failed to set network %s's openflow ports for secondary bridge; error: %v", udng.GetNetworkName(), err)
+				}
+			}
+			return true, nil
+		}
+		postFunc := func() error {
+			if err := udng.Reconcile(); err != nil {
+				return fmt.Errorf("failed to reconcile flows on bridge for network %s; error: %v", udng.GetNetworkName(), err)
+			}
+			return nil
+		}
+		waiter.AddWait(readyFunc, postFunc)
+		if err := waiter.Wait(); err != nil {
+			return err
+		}
+	}
+	return nil
 }
 
 // DelNetwork will be responsible to remove all plumbings
 // used by this UDN on the gateway side
 func (udng *UserDefinedNetworkGateway) DelNetwork() error {
+	if udng.openflowManager != nil {
+		udng.openflowManager.delNetwork(udng.NetInfo)
+		// TODO (dceara): Do we need to reconcile all flows? Is this heavy?
```

Comment:
did you manage to answer this concern? Is there follow on work needed to optimise this issue?

Reply:
It really depends on how many UDNs we expect to support.  Short answer: no, not yet. :)

Reply:
yea I spoke to @trozet about this.. he mentioned he is not concerned around this because flows are atomically added and deleted....

Reply:
if tim agrees then I could remove the comment

Reply:
Removed it in latest commit. Still need Tims ack.

Reply:
It's not heavy for openflow reconciliation. We just update the OF cache and then next time (in 5 seconds or next time its requested) OF flows will be atomically updated. Reconcile however does ask for all services to be requeued for nodePortWatcher to process every key and then also update openflow flows.

If a network gets deleted, we have to update all the flows including service flows, so I dont see anyway around this. I don't think it will be that bad, other than nodePortWatcher may increase its workload if lots of networks are being created/deleted and there are a lot of services in the cluster.

Reply:
resolving this comment...we can do scale tests and improve from there if we see slowness.